### PR TITLE
PERF-#4929: Compute `dtype` when using `Series.dt` accessor

### LINKED ISSUE #4929: PERF: compute `dtype` when using `Series.dt` accessor
In most cases, the type is known in advance.

### PR DESCRIPTION
**Pull request by anmyachev**
Signed-off-by: Myachev <anatoly.myachev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/development/contributing.html#commit-message-formatting)
- [x] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #4929 
- [x] tests added and passing
- [x] module layout described at `docs/development/architecture.rst` is up-to-date 
- [x] added (Issue Number: PR title (PR Number)) and github username to release notes for next major release

### CODE REVIEW

**Review by pyrito — COMMENTED** [review]
How much speedup are we getting with this change?

**Comment by anmyachev** [comment]
> How much speedup are we getting with this change?

These changes improve asynchronous behavior when the next operation needs to know the types.

File: modin/core/storage_formats/pandas/query_compiler.py
Comment (vnlitvinov) [inline modin/core/storage_formats/pandas/query_compiler.py:1451]: could you please check this is correct? I mean, wasting int64 to store numbers from 0 to 59 feels a little too much...

Comment (pyrito) [reply modin/core/storage_formats/pandas/query_compiler.py:1451]: @vnlitvinov agree that it does seem a bit wasteful.

File: modin/core/storage_formats/pandas/query_compiler.py
Comment (jbrockmendel) [reply modin/core/storage_formats/pandas/query_compiler.py:1445]: ser.dt is available on Series _with_ datetime64 dtypes, the ser.dt.foo methods here return the dtypes specified

File: modin/core/storage_formats/pandas/query_compiler.py
Comment (jbrockmendel) [reply modin/core/storage_formats/pandas/query_compiler.py:1451]: this could probably be improved upstream.  the lower-level calls return int32, get wrapped somewhere in int64

Comment (anmyachev) [reply modin/core/storage_formats/pandas/query_compiler.py:1451]: I checked using the following example, everything is so. Do you need to do something else or we can merge?
```python
pd.Series(pd.to_timedelta(np.arange(5), unit='d')).dt
```

**Review by vnlitvinov — APPROVED** [review]
LGTM, thanks!

**Comment by anmyachev** [comment]
@vnlitvinov could you merge it or you wait for someone' review?

**Comment by anmyachev** [comment]
@pyrito do you know how to restart docs CI build?

**Comment by pyrito** [comment]
> @pyrito do you know how to restart docs CI build?

The only way I've been able to rerun is to amend the commit and force push to retrigger everything.

**Comment by vnlitvinov** [comment]
> @vnlitvinov could you merge it or you wait for someone' review?

I usually try to give at least a day after my approval to let someone else report any issues they see with a PR.
Seems that no one is objecting this time, so merging 😺

**Comment by anmyachev** [comment]
> > @vnlitvinov could you merge it or you wait for someone' review?
> 
> I usually try to give at least a day after my approval to let someone else report any issues they see with a PR. Seems that no one is objecting this time, so merging 😺

I see, thanks!

**Pull request merged**
